src/Promitor.Integrations.LogAnalytics/LogAnalyticsClient.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Monitor.Query;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Extensions.Logging;
+
+namespace Promitor.Integrations.LogAnalytics
+{
+    public class LogAnalyticsClient
+    {
+        private readonly ILogger _logger;
+        public readonly string ColumnNameResult = "result";
+        private readonly Uri _defaultEndpoint = new Uri("https://api.loganalytics.io");
+        private readonly Uri _govEndpoint = new Uri("https://api.loganalytics.us");
+
+        private readonly LogsQueryClient _logsQueryClient;
+
+        public LogAnalyticsClient(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<LogAnalyticsClient>();
+            _logsQueryClient = new LogsQueryClient(new DefaultAzureCredential());
+        }
+
+        public LogAnalyticsClient(ILoggerFactory loggerFactory, AzureEnvironment azureEnvironment, TokenCredential tokenCredentials)
+        {
+            _logger = loggerFactory.CreateLogger<LogAnalyticsClient>();
+            var uri = _defaultEndpoint;
+            if (azureEnvironment.Name == nameof(AzureEnvironment.AzureUSGovernment))
+                uri = _govEndpoint;
+
+            _logsQueryClient = new LogsQueryClient(uri, tokenCredentials);
+        }
+
+        public async Task<double> QueryDouble(string workspaceId, string query, TimeSpan aggregationInterval)
+        {
```

Comment:
Let's add guards for the input

Reply:
Not sure how to guard the struct TimeSpan

Reply:
Guard.For(() => aggregationInterval == default)

Reply:
I think it needs to be Guard.For(() => aggregationInterval != default). The default for timeSpan is 00:00:00:00